apis/shared-dimensions/lib/domain/shared-dimension.ts

```diff
@@ -29,6 +29,10 @@ export async function create({ resource, store }: CreateSharedDimension): Promis
     throw new DomainError('Missing dimension identifier')
   }
 
+  if (identifier === 'hierarchy') {
+    throw new DomainError('`hierarchy` is not an allowed shared dimension identifier')
+  }
```

Comment:
Since the decision to rewrite is based on the URIs, I added an additional restriction so that one cannot create a shared dimension called `/hierarchy`. That would interfere with the [`rewriteSharedDimensionTerms` function](https://github.com/zazuko/cube-creator/pull/1185/files#diff-7e20c4d710b7de02ee4a6072fbc7e09077f2b2d9020235d2857d14fb829de9e9R10)